overlay.py

```diff
@@ -169,7 +175,13 @@ def _on_connect(self, client, userdata, flags, rc):
         self.client.subscribe(str(DAShboard.recording))
         with self.exception_handler:
             self.on_connect(client, userdata, flags, rc)
-        self.backend.on_base_canvas_updated(self.base_canvas)
+        print("Connected with rc: {}".format(rc))
+
+    def on_connect(self, client, userdata, flags, rc):
```

Comment:
Based on what Chris said https://github.com/monash-human-power/raspicam/pull/48#discussion_r491239207, this shouldn't be abstract for the moment. Since `overlay_new.py` does not override `on_connect()` because it doesn't need to anymore. However, this is currently used in the other overlays. The other overlays currently use `on_connect()` to draw the base layer. 

I imagine in the future, we will need to update the other overlays to start using `draw_base_layer()`. However, I don't think this is appropriate for this PR, specifically because the other overlays are very outdated compared to `overlay_new`. This should be a PR on its own, since the other overlays don't seem to be using `components`, and they definitely need a lot of refactoring. Combining those tasks would make this PR too unnecessarily big, in my opinion.

Reply:
Will definitely bring this up tomorrow with Chris and suggest making a PR to update the other overlays.

Reply:
Based on what Chris said in https://github.com/monash-human-power/raspicam/pull/48#discussion_r491239207, this shouldn't be abstract for the moment, because this would force the subclass overlays to implement this when they no longer need it. At the moment, the only overlay that uses `on_connect()` is `overlay_error` which is used for testing whether an error message will be sent to the DAShboard. 

I definitely think there needs to be a lot of refactoring done with the Raspicam, because of all these new changes. I will talk to Chris tomorrow about cleaning up a lot of the code in the other overlays, because I really love how `overlay_new` currently handles displaying components onto the overlay.

Reply:
Upon saying that, I won't do this in this PR now, because that would be too big and would need to be a PR on its own. Hopefully you understand.

Reply:
Yep my bad I didn't seem Chris's comment regrading this.

Reply:
Haha all good, I resolved it so there was no way you could've seen that. Not your fault 😄